uktrade/data-hub-frontend
# Remove integration with the Data Store service

## Description of change

The Data Store service (which we use in the add company form for matching postcodes to UK regions) is being retired. In order to preserve this functionality, I've created a mapping function and verified the UK region for each postcode against the Data Workspace region dataset.

This PR will allow the service to be retired without reducing the user experience.

<details>
<summary>Some additional implementation details</summary>

UK postcodes can be broken down into several components.

![uk-postcode-components](https://github.com/uktrade/data-hub-frontend/assets/36161814/a44b4a2a-31c4-433e-891e-e245dea186e2)

The section associated with the UK region is the **area**. This can safely be extracted by initially removing any spaces, then removing the incode (this is always three characters long).

The outcode itself has two segments - the area and the district. The district will always start with a district number, so the next stage of this process is to identify the first number in the string and remove it (as well as all subsequent characters). This leaves the area.

Below is an example of the above process using the postcode in the image:
- SW1A 2AA
- SW1A2AA (removing spaces)
- SW1A (removing incode)
- SW (removing district) 

</details>

## Test instructions

Manually add a UK company to Data Hub, and enter a valid postcode in the address field. The region should automatically be selected on the next page. It should also be possible to add an invalid postcode in the address field and submit the company with that postcode.

There should be no other changes to the add company journey.

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
